components/skus/browser/skus_sdk_fetcher_impl.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/skus/browser/skus_sdk_fetcher_impl.h"
+
+#include <utility>
+#include <vector>
+
+#include "brave/components/skus/browser/rs/cxx/src/lib.rs.h"
+#include "net/base/load_flags.h"
+#include "url/gurl.h"
+
+namespace {
+
+// Maximum accepted size of response from SKU SDK server. 1MB.
+const int kMaxResponseSize = 1024 * 1024;
+
+}  // namespace
+
+namespace skus {
+
+SkusSdkFetcherImpl::SkusSdkFetcherImpl(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : url_loader_factory_(url_loader_factory) {}
+
+SkusSdkFetcherImpl::~SkusSdkFetcherImpl() {}
+
+void SkusSdkFetcherImpl::BeginFetch(
+    const skus::HttpRequest& req,
+    rust::cxxbridge1::Fn<
+        void(rust::cxxbridge1::Box<skus::HttpRoundtripContext>,
+             skus::HttpResponse)> callback,
+    rust::cxxbridge1::Box<skus::HttpRoundtripContext> ctx) {
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = GURL(static_cast<std::string>(req.url));
+  resource_request->method = static_cast<std::string>(req.method).c_str();
+  resource_request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+  // No cache read, always download from the network.
+  resource_request->load_flags =
+      net::LOAD_BYPASS_CACHE | net::LOAD_DISABLE_CACHE;
+
+  for (size_t i = 0; i < req.headers.size(); i++) {
+    resource_request->headers.AddHeaderFromString(
+        static_cast<std::string>(req.headers[i]));
+  }
+
+  sku_sdk_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), GetNetworkTrafficAnnotationTag());
+
+  sku_sdk_loader_->DownloadToString(
+      url_loader_factory_.get(),
+      base::BindOnce(&SkusSdkFetcherImpl::OnFetchComplete,
+                     base::Unretained(this), std::move(callback),
+                     std::move(ctx)),
+      kMaxResponseSize);
```

Comment:
`constexpr size_t kMaxResponseSizeDefault = 4194304;  // 4 * 1024 * 1024` is this always going to be the case that it is smaller?

Reply:
if not we should be using `DownloadToStringOfUnboundedSizeUntilCrashAndDie`

Reply:
Fixed with https://github.com/brave/brave-core/pull/10847/commits/a053bc6de9fe494884f3fdf8d8c9bc993e4a184e